src/BatteryStateOfChargeService.h

```diff
@@ -17,6 +17,25 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
 private:
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
-
+    double objectCurrent;
+    double inCurrent=0;
```

Comment:
So when you want to initialize values, you should do it in the constructor. Having them initialized here is unclear and also doesn't compile on some compilers! On the compiler that comes with Qt, you get a warning.